src/mpid/ch4/shm/xpmem/xpmem_send.h

```diff
@@ -12,52 +12,101 @@
 #include "xpmem_pre.h"
 #include "xpmem_impl.h"
 
-MPL_STATIC_INLINE_PREFIX int MPIDI_XPMEM_lmt_isend(const void *buf, MPI_Aint count,
-                                                   MPI_Datatype datatype, int rank, int tag,
-                                                   MPIR_Comm * comm, int context_offset,
-                                                   MPIDI_av_entry_t * addr, MPIR_Request ** request)
+MPL_STATIC_INLINE_PREFIX int MPIDI_XPMEM_lmt_rts_isend(const void *buf, MPI_Aint count,
+                                                       MPI_Datatype datatype, int rank, int tag,
+                                                       MPIR_Comm * comm, int context_offset,
+                                                       MPIDI_av_entry_t * addr,
+                                                       MPIR_Request ** request)
 {
     int mpi_errno = MPI_SUCCESS;
     MPIR_Request *sreq = NULL;
+    MPIDI_SHM_ctrl_hdr_t ctrl_hdr;
     size_t data_sz;
     MPI_Aint true_lb;
-    bool is_contig ATTRIBUTE((unused)) = 0;
-    MPIDI_SHM_ctrl_hdr_t ctrl_hdr;
-    MPIDI_SHM_ctrl_xpmem_send_lmt_req_t *slmt_req_hdr = &ctrl_hdr.xpmem_slmt_req;
```

Comment:
Can you please clean up unrelated changes? E.g., moving around `MPIDI_SHM_ctrl_hdr_t ctrl_hdr;` and `bool is_contig`. 

Besides, you should not change `bool is_contig ATTRIBUTE((unused)) = 0;`. The parameter is not used when build with --enable-fast because `MPIR_Assert` becomes a no-op.